testing/bin/run_tests.py

```diff
@@ -76,27 +99,49 @@ def main():
     parser.add_argument("--it", action="store_true", help="Restrict scope to integration tests.")
     options, passthrough_args = parser.parse_known_args()
 
-    coloredlogs.install(level="INFO", fmt="%(levelname)s %(message)s", isatty=options.color)
+    coloredlogs.install(
+        level=options.log_level, fmt="%(levelname)s %(message)s", isatty=options.color
+    )
     logger = logging.getLogger(parser.prog)
+    logger.log(
+        logging.root.level, "Logging configured at level {level}.".format(level=options.log_level)
+    )
 
     if options.devpi:
         if options.shutdown_devpi:
             atexit.register(devpi.shutdown)
         launch_result = devpi.launch(
+            host=options.devpi_host,
             port=options.devpi_port,
             timeout=options.devpi_timeout,
             max_connection_retries=options.devpi_max_connection_retries,
             request_timeout=options.devpi_request_timeout,
         )
-        if not launch_result:
-            logger.warning("Failed to launch devpi server. Continuing without it...")
-        else:
+        if isinstance(launch_result, devpi.LaunchResult):
             os.environ["_PEX_TEST_DEFAULT_INDEX"] = launch_result.url
+            os.environ["PIP_INDEX_URL"] = launch_result.url
```

Comment:
This was a leak I noticed when fixing one of the broken tests in a later commit in this PR. Some test code uses `pip install`, etc. directly and that was still going out to PyPI.